lib/apiGateway/schema.js

```diff
@@ -168,6 +243,34 @@ const proxiesSchemas = {
       queueName: stringOrGetAtt('queueName', 'QueueName').required(),
       requestParameters
     })
+  }),
+  dynamodb: Joi.object({
+    dynamodb: proxy.append({
+      action: Joi.string()
+        .valid(allowedDynamodbActions)
+        .when('method', {
+          is: Joi.string()
+            .valid(['options', 'head', 'any'])
+            .insensitive(),
+          then: Joi.required().error(
+            customErrorBuilder(
+              'any.required',
+              '"action" is required when you define options, head, any to "method" property'
```

Comment:
can we say `"action" is required when "method" is "options", "head", or "any"` instead? reads better (to me at least!)